src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java

```diff
@@ -649,7 +667,37 @@ public void messageArrived(String topic, MqttMessage message) throws Exception {
                 if ("@set".equals(action)) {
                     location.pop();
                     set(topic, location, messageFormat, data);
-                } else {
+                }
+                else if("@exec".equals(action)){
+                    JobsParameter jobsParameter = new JobsParameter();
+
+                    messageFormat.deserializeJobsParameter(data, jobsParameter);
+
+                    JobsManager.getInstance().executeJob(jobsParameter.getJobURI(),jobsFilePath
+                            , jobsParameter.getGetOuput(), internal.mqtt, internal.uuid);
+
+                }
+                else if("@logsLevel".equals(action)){
+                    System.out.println("loglevel set");
+                    LogParameter logParameter = new LogParameter();
+
+                    messageFormat.deserializeLogParameter(data, logParameter);
+
+                    try{
+                        Level log4jLevel = Level.getLevel(logParameter.getLevel());
+                        System.out.println(log4jLevel.toString());
+                        Configurator.setRootLevel(log4jLevel);
+                    }catch (Exception e){
+                        log.error("Level \"" + logParameter.getLevel() + "\" not supported!");
+                        e.printStackTrace();
+                        //}
+                    }
+                }
+                else if("@logsLevelUnRetained".equals(action)){
```

Comment:
This is an action in order the endpoint publishes his log level, right? I think the name is misleading, especially as the message is send retained.

Reply:
Nop. I'm using two topic to send the log level (logsLeve, logsLevelUnRetained)  and I use the retain of mqtt to make the endpoint get it's loglevel when it start up.
Sadly amqp doesn't support retained... and the rest api will send a message using amqp to the broker. So I send it to logsLevelUnRetained and the endpoint will send it to itself again with the retain flag.

Reply:
Ok, I see. The problem with that is that if the server gets restarted, all retained messages will be gone. Maybe we could use the PAHO persistent storage API to save that value?

Reply:
Another solution, since the log level is stored in mongo, is to send a setLevel to the endpoint in 'answer' to the @online? Make more sense in fact no?

Reply:
Last commit include the persistence file we talked about